syntax = "proto3";

package aa.shared;

import "shared/proto/frame.proto";

option cc_enable_arenas = true;

// Processing request
message ProcessRequest {
  VideoFrame frame = 1;
  string operation = 2;  // e.g., "blur", "edge_detect", "resize"
  map<string, string> parameters = 3;
}

// Processing response
message ProcessResponse {
  VideoFrame result = 1;
  bool success = 2;
  string error_message = 3;
  double processing_time_ms = 4;
}

// Health check
message HealthRequest {}

message HealthResponse {
  bool healthy = 1;
  string status = 2;
}

// Video processing service
service VideoProcessingService {
  rpc ProcessFrame(ProcessRequest) returns (ProcessResponse);
  rpc HealthCheck(HealthRequest) returns (HealthResponse);
  rpc StreamProcess(stream ProcessRequest) returns (stream ProcessResponse);
}
